server/statshouse/statshouse-client.cpp

```diff
@@ -50,18 +101,26 @@ bool StatsHouseClient::init_connection() {
   return true;
 }
 
-void StatsHouseClient::send_metrics(const std::vector<StatsHouseMetric> &metrics) {
+void StatsHouseClient::send_metrics() {
   if (port == 0 || (sock_fd <= 0 && !init_connection())) {
     return;
   }
-  StatsHouseAddMetricsBatch metrics_batch = {.fields_mask = vk::tl::statshouse::add_metrics_batch_fields_mask::ALL, .metrics = metrics};
-  int stored_size = vk::tl::store_to_buffer(buffer, BUFFER_LEN, metrics_batch);
-  assert(stored_size < BUFFER_LEN);
 
-  ssize_t len = send(sock_fd, buffer, stored_size, 0);
-  if (len < 0) {
-    log_server_error("Can't send metrics to statshouse: %s", strerror(errno));
+  auto [result, len] = prepare_statshouse_stats(statshouse_stats_t{tags}, "kphp", STATS_TAG_KPHP_SERVER);
+
+  ssize_t slen = send(sock_fd, result, len, 0);
+  if (slen < 0) {
+    log_server_error("Can't send metrics to statshouse (len = %i): %s", len, strerror(errno));
+  }
+}
+
+StatsHouseClient::StatsHouseClient() {
+  char hostname[128];
+  int res = gethostname(hostname, 128);
```

Comment:
Is it really good practice to call this functions in a constructor of a global singleton? @DrDet

Reply:
It seems quite reasonable here. What confuses you?

Reply:
I don't understand, when this constructor will be called. Is there a chance, that `gethostname()` hangs, and master process will hang before `main()`?

Reply:
It will be called on the first `vk::singleton<...>::get()` call which happens on parsing host:port option. I think it never hangs because it doesn't use DNS, I think it just reads some file